### ISSUE TITLE
IDA 7.2 bug due to calling function from non-main thread

### ISSUE THREAD

**Issue by joseph-hannon** [alexhude/uEmu#12]
**Issue**
When running uEmu as a plugin in IDA 7.2 and stepping over code I occasionally get the error:
```
Traceback (most recent call last):
  File "_ctypes/callbacks.c", line 315, in 'calling callback function'
  File "C:\python27-x64\lib\site-packages\unicorn\unicorn.py", line 448, in _hook_mem_access_cb
    cb(self, access, address, size, value, data)
  File "C:/Program Files/IDA 7.2/plugins/uEmu.py", line 1169, in hook_mem_access
    if self.is_breakpoint_reached(address):
  File "C:/Program Files/IDA 7.2/plugins/uEmu.py", line 1422, in is_breakpoint_reached
    for idx in range(IDAAPI_GetBptQty()):
  File "C:\Program Files\IDA 7.2\python\idc.py", line 5796, in get_bpt_qty
    return ida_dbg.get_bpt_qty()
  File "C:\Program Files\IDA 7.2\python\ida_dbg.py", line 1907, in get_bpt_qty
    return _ida_dbg.get_bpt_qty(*args)
RuntimeError: Function can be called from the main thread only
```

**Cause**
I think it's because IDA 7.2 [introduced checks](https://www.hex-rays.com/products/ida/7.2/index.shtml) for IDA API functions to make sure they can only be called from the main thread. 

> IDAPython: all functions not marked as THREAD_SAFE in the C++ SDK, will now check that they are being called from the main thread, avoiding possible corruption or crashes

**Comment by alexhude** [comment]
this issue is already addressed in upcoming commit

**Comment by alexhude** [comment]
Fixed in #0fa6c62
